### PR TITLE
Allow an empty payload in send()

### PR DESCRIPTION
**Pull request by mcollina**
This fixes the reason why #70 is failing on node v7.

### CODE REVIEW

File: lib/reply.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ Reply.prototype.send = function (payload) {
     this.res.setHeader('Content-Type', 'application/json')
 
     // Here we are assuming that the custom serializer is a json serializer
-    const str = this._serializer ? this._serializer(payload) : serialize(this.handle, payload)
+    const str = (this._serializer ? this._serializer(payload) : serialize(this.handle, payload)) || ''
```
Comment (delvedor) [inline lib/reply.js]: With empty content the code should be 204, we are already supporting that [here](https://github.com/fastify/fastify/blob/master/lib/reply.js#L44-L46), but we are not handling it correctly (in other words, we are not closing the response).
I think we should handle it all there and not here. What do you think?

Comment (mcollina) [inline lib/reply.js]: that's not the case, because I set the code in `reply.code(200).send()`.

Comment (delvedor) [inline lib/reply.js]: I don't understand, why not?
What I mean is:
```js
if (!payload) {
  if (!this.res.statusCode) {
    this.res.statusCode = 204
  }
  setImmediate(wrapReplyEnd, this, '')
  return
}
```

**Comment by mcollina** [comment]
@delvedor updated with your suggestions.

**Pull request merged**